cluster/main.tf

```diff
@@ -1,31 +1,142 @@
-  max_amount_workers = "${var.max_amount_workers}"
-  k8s_data_bucket    = "${var.k8s_data_bucket}"
-  cluster_cidr       = "${var.cluster_cidr}"
-  k8s_version        = "${var.k8s_version}"
-}
-
-module "masters" {
-  source         = "../master"
-  ami            = "${var.master_ami}"
-  instance_type  = "${var.master_instance_type}"
-  key_name       = "${var.key_name}"
-  bastion_sg     = "${var.bastion_sg}"
-  subnets        = ["${var.master_subnets}"]
-  project        = "${var.project}"
-  environment    = "${var.environment}"
-  amount_masters = "${var.amount_masters}"
-  k8s_data_bucket = "${var.k8s_data_bucket}"
-  cluster_cidr       = "${var.cluster_cidr}"
-  k8s_version        = "${var.k8s_version}"
-  service_ip_range   = "${var.service_ip_range}"
+terraform {
+	required_version = "> 0.8.0"
 }
+
+# TODO list:
+#  * Allow configuring the API ELB type: Public or Internal
+#  * Use separate management and node subnets once the fix for this is released:
+#      https://github.com/kubernetes/kops/issues/1980
+#
+### NOTE: Do not change the layout of the template files or your will get unnecessary empty lines
+###       in the generated output.
+
+data "aws_vpc" "vpc_for_k8s" {
+  id = "${var.vpc_id}"
+}
+
+data "aws_availability_zones" "available" {
+  state = "available"
+}
+
+#########################################################
+# Subnets
+#########################################################
+
+data template_file "master-subnet-spec" {
+  count = "3"
+  template = "${file("${path.module}/../templates/kops-cluster-subnet.tpl.yaml")}"
+
+  vars {
+    name = "${element(formatlist("master-%s", data.aws_availability_zones.available.names),count.index)}"
+    type = "Private"
+    zone = "${element(data.aws_availability_zones.available.names, count.index)}"
+    id   = ""
+    cidr = "${cidrsubnet(data.aws_vpc.vpc_for_k8s.cidr_block,8,var.master_net_number+count.index)}"
+  }
+}
+
+# Private subnets to host the Kubernetes worker nodes
+# Currently disabled due to https://github.com/kubernetes/kops/issues/1980
+# data template_file "worker-subnet-spec" {
+#   count = "3"
+#   template = "${file("${path.module}/../templates/kops-cluster-subnet.tpl.yaml")}"
+
+#   vars {
+#     name = "element(formatlist("worker-%s", data.aws_availability_zones.available.names)"
+#     type = "Private"
+#     zone = "${element(data.aws_availability_zones.available.names, count.index)}"
+#     id   = ""
+#     cidr = "${cidrsubnet(data.aws_vpc.vpc_for_k8s.cidr_block,8,var.worker_net_number+count.index)}"
+#   }
+# }
+
+# Utility subnets are public for ELB creation.
+data template_file "utility-subnet-spec" {
+  count = "3"
+  template = "${file("${path.module}/../templates/kops-cluster-subnet.tpl.yaml")}"
+
+  vars {
+    name = "${element(formatlist("utility-%s", data.aws_availability_zones.available.names),count.index)}"
+    type = "Utility"
+    zone = "${element(data.aws_availability_zones.available.names, count.index)}"
+    id   = ""
```

Comment:
also this will always be empty no? shouldn't we make this a variable or remove it at all?